### PR TITLE
Adds egging to transformation.

### PR DESCRIPTION
**Pull request by killer653**
# **FULLY WORKING AND READY FOR MERGE**

Addresses issue #444 on most parts.

This allows you to egg tf crew members.

Egg sprites: http://puu.sh/nEMXk/232f074249.png (UPDATED)

**Features**:
Two egg sprites for each egg., one torn open, one closed.
Takes two minutes for the prey to break out of the egg
Anyone outside the egg can easily tear it open (Click on the egg twice to tear it open.)
Egg can be destroyed by using a welder.
Has all the regular TF modes (male, female, keep gender, change species)

Custom eggs for almost every species (Custom species are included with whatever their "base" species in. For example, a Dragon that has the Unathi as a base will show up as "Dragon Unathi egg"
Note: I tried every viable solution to fix this problem, but the custom_species variable isn't "null", but rather " " (space), meaning I can't put it in to check for it without causing compile errors. This was the only other solution other than scrapping the project altogether. I physically can not improve it anymore than it currently is.

Code is **FULLY** tested. Everything is working as expected with no bugs.

Limitations:
As listed above, custom species will piggy back off of their main species. This can be slightly immersion breaking, as it will say "Dragon Unathi egg" instead of "Dragon egg", but it's impossible to fix this, as the variable isn't "null", but rather " " (space), which would cause compile errors if I looked for " ".

I physically can not improve it any more than it currently is. I'm sorry, but it's virtually impossible for me to fix it. I left in the commented out code so someone can uncomment it and fiddle around with it and see if they can find any way to fix it, but I just can't do it. I'm under the assumption that people would much rather deal with an occasional out of place word than not having the egging feature at all. There's no current bugs that have been found, and I did extensive testing on it.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
I don't want to merge a prototyped feature, and I see a lot of commented out code for different species' eggs.

**Comment by killer653** [comment]
It's less of a prototype. It's a fully functioning feature currently, and everything works. The commented out eggs are for further expansion, so if someone wants to resprite the eggs for the other races/add an addition to the code so the species have specific eggs, then it's there for them to easily access.

**Comment by SpadesNeil** [comment]
> "That is an Unathi egg. Some species of Unathi apparently lay soft-shelled eggs!"
> "That is a Nevarean egg. Most Nevareans lay hard-shelled eggs!"
> "That is a Human egg. Some humans--wait, what?"
> "That is a Tajaran egg. Apparently that's what a Tajaran egg looks like. Weird."
> "That is a Skrell egg. It's soft and squishy."
> "That is a [custom_species] egg. It has a very unique look to it."

We're missing this.

**Comment by Cameron653** [comment]
That's where I kind of had a problem. I _could_ possibly make it check to see if they're a certain race and, if so, make it let them pick that specific egg, but I'm kind of stuck on _how_ to exactly do that. I was thinking of maybe doing a check to see if their species = (insert species) here, and if it's X species, make it spawn in X egg. The problem comes in that I'd have to make variables for each one of the eggs, which is kind of a problem, as it'd spawn X ammount of eggs that i make X ammount of varaibles for.

Another idea I had was making it so it would have them all select able on the transformation menu, which wouldn't be too hard to code, but it'd heavily clog up the  transformation menu, as you'd need 20 egg settings for all five of the races, and that's not accounting for custom races.

That being said, I do see a _possible_ way to make all those eggs work without causing major bugs. Going to come back to this in a bit and do some more work on it.

**Comment by antsnap** [comment]
Seem reasonable to merge. I will add though, that one of the more common transform-fetish-doing races is the Slime Person, so if we got a slime egg or something that'd be super neato.

File: code/game/objects/structures/crates_lockers/closets/egg.dm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/obj/structure/closet/secure_closet/egg/shark
+	name = "akula egg"
+	desc = "It's soft and slimy to the touch"
+	icon = 'icons/obj/closet.dmi'
+	icon_closed = "egg_akula"
+	icon_opened = "egg_akula_open" 
+	
+/obj/structure/closet/secure_closet/egg/sergal
+	name = "sergal egg"
+	desc = "An egg with a slightly fuzzy exterior, with a hard layer beneath."
```
Comment (SpadesNeil) [inline code/game/objects/structures/crates_lockers/closets/egg.dm]: Change to:
> An egg with a slightly fuzzy exterior, and a hard layer beneath.

Grammar.

File: code/game/objects/structures/crates_lockers/closets/egg.dm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+/obj/structure/closet/secure_closet/egg/skrell
+	name = "skrell egg"
+	desc = "It's soft and squishy"
+	icon_state = "egg_skrell"
+	icon_closed = "egg_skrell"
+	icon_opened = "egg_skrell_open" 
+
+/obj/structure/closet/secure_closet/egg/shark
+	name = "akula egg"
+	desc = "It's soft and slimy to the touch"
```
Comment (SpadesNeil) [inline code/game/objects/structures/crates_lockers/closets/egg.dm]: Punctuation!

File: code/game/objects/structures/crates_lockers/closets/egg.dm [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+/obj/structure/closet/secure_closet/egg/tajaran
+	name = "tajaran egg"
+	desc = "Apparently that's what a Tajaran egg looks like. Weird." 
+	icon_state = "egg_tajaran"
+	icon_closed = "egg_tajaran"
+	icon_opened = "egg_tajaran_open" 
+	
+/obj/structure/closet/secure_closet/egg/skrell
+	name = "skrell egg"
+	desc = "It's soft and squishy"
```
Comment (SpadesNeil) [inline code/game/objects/structures/crates_lockers/closets/egg.dm]: Punctuation!

File: code/game/objects/structures/crates_lockers/closets/egg.dm [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+/obj/structure/closet/secure_closet/egg/nevrean
+	name = "nevarean egg"
+	desc = "Most Nevareans lay hard-shelled eggs!"
+	icon_state = "egg_nevarean"
+	icon_closed = "egg_nevarean"
+	icon_opened = "egg_nevarean_open" 
+	
+/obj/structure/closet/secure_closet/egg/human
+	name = "human egg"
+	desc = "Some humans--wait, what?"
```
Comment (SpadesNeil) [inline code/game/objects/structures/crates_lockers/closets/egg.dm]: I know I made this line but it looks bad to me now. Change to:
> Some humans lay eggs that are--wait, what?

**Comment by SpadesNeil** [comment]
Original species eggs also need an egg that is renamed according to what species laid it.

For example, if my custom species is Deathclaw, it should be a "Deathclaw egg."

The description could simply be:
"This egg has a very unique look to it."

**Comment by killer653** [comment]
Going to be fixing these errors when I get home to my primary PC. I really can't do any testing right now, and this code, in it's current state, is still _quite_ flawed.

Do NOT merge, or you might get compile errors. I'm still actively working on the code.

**Comment by killer653** [comment]
I've improved it as best as I can. I've tried multiple methods, but I found it's virtually impossible to make it check for a custom species. The variable is set to "null" in the code, but actually testing this in game shows that it is rather just a space, which would be put into the code as " ", which causes compile errors. 

I've bypassed this by making it follow this format:
(Custom_Species) (Species) egg
If a person doesn't have a custom species, and they're an unathi, it'll show up as Unathi egg.
If a person has "Dragon" and is an unathi, the egg will show up as "Dragon Unathi egg", which can be slightly immersion breaking, but it's virtually impossible to fix this problem, and it's way out of my grasp

All the sprites are working, all of the mechanics are working, but I just can't fix the "special egg". I've done as much as I can. I'm assuming people would much rather put up with one occasional out of place word, than not having the feature at all. I'm really sorry about being unable to fix it, but I've coded it to the best of my abilities.

**Comment by SpadesNeil** [comment]
Why not store both species in a temporary var?

Then you can say instead of
>(Custom_Species) (Species) egg

you can go

> (defined_species) egg

and before that, you go

if (custom_species)
defined_species = usr.custom_species
else
defined_species = usr.species

**Comment by killer653** [comment]
I'm going to try that out. See if I can perhaps get it to work.

**Comment by killer653** [comment]
# **Fully working and ready for merge**
It's done!
I made it so that if it's a special species, like "Dragon", then it changes the description to 
"This egg has a very unique look to it."
which works. Adding to that, it also has the species specific egg sprite.

# **Fully working and ready for merge**

**Comment by optimumtact** [comment]
Squash your commits

**Comment by Cameron653** [comment]
I know, I'm an awful coder. I would squash it down, but most of the coding I did was on the github website, instead of the desktop client, so you can't really combine it into one big commit.

**Pull request merged**